software/bsg_manycore_lib/bsg_manycore.h

```diff
@@ -147,38 +147,41 @@ inline void bsg_fence()      { __asm__ __volatile__("fence" :::); }
 //------------------------------------------------------
 
 // Pointers to remote locations (non-scratchpad) could be qualified
-// with __remote. Doing so would tell compiler to assign a different
+// with bsg_attr_remote. Doing so would tell compiler to assign a different
 // address space to the contents of those pointers and latencies of
 // memory accesses would be considered as 20 cycles. For example,
-// `__remote float* foo;` essentially declares foo as `__remote float*`
+// `bsg_attr_remote float* foo;` essentially declares foo as `bsg_attr_remote float*`
 // type, and the compiler assumes loads from `foo` to take 20 cycles
 // on average.
 #ifdef __clang__
-#define __remote __attribute__((address_space(1)))
+#define bsg_attr_remote __attribute__((address_space(1)))
+#elif defined(__GNUC__) && !defined(__cplusplus)
+// TODO: Define the following with correct keyword for GCC
+#define bsg_attr_remote
```

Comment:
__remote is what you want

Reply:
That worked... even made slight improvement in the codegen with this testcase!